insights/configtree/iniconfig.py

```diff
@@ -0,0 +1,89 @@
+"""
+This module models ini configuration as a tree.
+
+A DSL is provided to query the tree through a select function or brackets [].
+The brackets allow a more conventional lookup feel but aren't quite as powerful
+as using select directly.
+"""
+import re
+
+from insights.configtree import Directive, DocParser, LineGetter, Section
+from insights.configtree import eq, first, last, typed
+
+
+class ConfigParser(DocParser):
+    def parse_directive(self, lg):
```

Comment:
Please add to `docs/api_index.rst` and add docstrings for documentation.  Here's an example diff to add all the missing modules:
```diff
diff --git a/docs/api_index.rst b/docs/api_index.rst
index f6b8531e..ce980560 100644
--- a/docs/api_index.rst
+++ b/docs/api_index.rst
@@ -1,6 +1,30 @@
 API Documentation
 =================
 
+insights.configtree
+-------------------
+
+.. automodule:: insights.configtree
+    :members:
+    :show-inheritance:
+    :undoc-members:
+
+insights.configtree.dictlike
+----------------------------
+
+.. automodule:: insights.configtree.dictlike
+    :members:
+    :show-inheritance:
+    :undoc-members:
+
+insights.configtree.iniconfig
+-----------------------------
+
+.. automodule:: insights.configtree.iniconfig
+    :members:
+    :show-inheritance:
+    :undoc-members:
+
 insights.core
 -------------
```